src/foam/core/templates.js

```diff
@@ -153,17 +116,112 @@ foam.CLASS({
         return g;
       }
     },
+    {
+      name: 'javaGrammar',
+      factory: function() {
+        var g = this.getAbstractGrammar();
+
+        var self = this;
+
+        g.addActions({
+          markup: function(v) {
+            var ret = '';
+            var wasSimple = self.simple;
+            for ( var i = 0; i < self.out.length; i++ ) {
+              if ( self.out[i].length == 1 && ( i == 0 || self.out[i-1].length > 1 ) ) {
+                ret += 'builder.append("';
+              }
+              ret += self.out[i];
+              if ( self.out[i].length == 1 && ( i == self.out.length -1 || self.out[i + 1].length > 1 ) )
+                ret += '");\n';
+            }
+            ret = wasSimple ? null : ret;
+            self.out = [];
+            self.simple = true;
+            return [wasSimple, ret];
+          },
+          'simple value': function(v) {
+            self.push("',\n self.",
+                v[1].join(''),
+                v[2],
+                ",\n'");
+          },
+          'raw values tag': function (v) {
+            var value = v[1].join('').replace(/\s/g, '');
+            var argExist = self.args.map(ar => ar.name).includes(value);
```

Comment:
Would be faster to create a property called argSet_ that contains the names of args, so you don't need to keep rebuilding this for every value.